compiler/src/dmd/expression.d

```diff
@@ -685,6 +685,18 @@ extern (C++) abstract class Expression : ASTNode
     Type type;      // !=null means that semantic() has been run
     Loc loc;        // file location
 
+    import dmd.root.optional;
+
+    /** 
+        The idea is to try and saved where an expression was lowered from.
+        This is for better error messages.
+     */
+    Optional!ASTNode origin;
```

Comment:
Silly question, `ASTNode` is a class, why does this need to be optional?

Reply:
I suppose it doesn't have to be but it's there firstly to be really obvious, and secondly because I was tempted to include some metadata (i.e. it could become a struct with an alias this without changing the rest of the code)

Reply:
its a silly question for a reason, if you have a reason that's fine.